src/neo/Ledger/TransactionRouter.cs

```diff
@@ -20,7 +20,7 @@ public TransactionRouter(NeoSystem system)
         protected override void OnReceive(object message)
         {
             if (message is not Preverify preverify) return;
-            system.Blockchain.Tell(new PreverifyCompleted(preverify.Transaction, preverify.Relay, preverify.Transaction.VerifyStateIndependent(system.Settings)), Sender);
+            system.Blockchain.Tell(new PreverifyCompleted(preverify.Transaction, preverify.Relay, preverify.Transaction.VerifyStateIndependent(system.Settings, system.StoreView.CreateSnapshot())), Sender);
```

Comment:
If it's state independant, we can't use the snapshot.

Reply:
As in the ApplicationEngine constructor class, if snapshot is null, it will use the default exec factor, which is wrong after exec factor on-chain changed. 
https://github.com/neo-project/neo/blob/1366e7a54c6681124914e02da3051da0dc188d60/src/neo/SmartContract/ApplicationEngine.cs#L127-L136

Reply:
After introduce the "exec_fee_factor" feature, we can hardly implement the "VerifyStateIndependent" method without snapshot. Do we really need this method?

Reply:
> After introduce the "exec_fee_factor" feature, we can hardly implement the "VerifyStateIndependent" method without snapshot. Do we really need this method?

That's the point... it's not `VerifyStateIndependent` ...

Reply:
> `VerifyStateIndependent`

Shall we remove `VerifyStateIndependent ` and take place of it with `VerifyStateDependent`? Seems `snapshot` is necessary for dynamic `exec_fee_factor ` Get.

Reply:
If `VerifyStateIndependent` need snapshot, we should remove it.